src/widgets/category/category-widget-model.js

```diff
@@ -55,16 +53,31 @@ module.exports = WidgetModel.extend({
     this.lockedCategories.reset([]);
   },
 
-  applyColors: function () {
-    this.set('isColorsApplied', true);
+  autoStyle: function () {
+    var style = this.autoStyler.getStyle();
+    if (!this.dataviewModel._dataProvider) {
+      this.dataviewModel.layer.set('cartocss', style);
+    } else {
+      var index = this.dataviewModel._dataProvider._layerIndex;
+      var sublayer = this.dataviewModel._dataProvider._vectorLayerView;
+      sublayer.setCartoCSS(index, style, true);
+    }
+    this.dataviewModel.set('autoStyle', true);
   },
 
-  cancelColors: function () {
-    this.set('isColorsApplied', false);
+  cancelAutoStyle: function () {
+    if (!this.dataviewModel._dataProvider) {
+      this.dataviewModel.layer.restoreCartoCSS();
+    } else {
+      var index = this.dataviewModel._dataProvider._layerIndex;
+      var sublayer = this.dataviewModel._dataProvider._vectorLayerView;
+      sublayer.renderers[index].restoreCartoCSS(true);
+    }
+    this.dataviewModel.set('autoStyle', false);
   },
 
-  isColorApplied: function () {
-    return this.get('isColorsApplied');
+  isAutoStyle: function () {
+    return this.dataviewModel.get('autoStyle');
```

Comment:
Could "autoStyle" be a property of the widgetModel? What do you think about using `autoStyled` (d at the end) and `isAutoStyled`?